CRM/Core/BAO/Setting.php

```diff
@@ -178,6 +178,9 @@ public static function setItem(
     /** @var \Civi\Core\SettingsManager $manager */
     $manager = \Civi::service('settings_manager');
     $settings = ($contactID === NULL) ? $manager->getBagByDomain($domainID) : $manager->getBagByContact($domainID, $contactID);
+    if (self::isEnvironmentSet($name, $value)) {
+      throw new api_Exception('CiviCRM Environment already set in civicrm.settings.php!');
```

Comment:
@eileenmcnaughton one requirement is to allow administrators to  specify in civicrm.settings.php whether the instance is running as production or non-production, since this file is generally different between prod and non-prod environments and in most workflows is not copied. But they should not be forced to specify it there. If it is specified there, however, then I don't see a workflow when one would want to override it via the browser, and have this override persist as the db is copied between instances. I know the regular pattern is to allow defaults set in civicrm.settings.php to be overridden, but that didn't seem to make sense in this use case.